src/RosClientNode.hpp

```diff
@@ -39,12 +41,20 @@ class RosClientNode : public QObject {
   template <typename T>
   void encode_ros_msg(
       const T& msg, const std::string& msg_type, const std::string& to_topic) {
-    flatbuffers::FlatBufferBuilder fbb;
-    auto metadata = encode_metadata(fbb, msg_type, to_topic);
-    auto root_offset = encode<T>(fbb, msg, metadata);
-    fbb.Finish(flatbuffers::Offset<void>(root_offset));
-    Q_EMIT ros_message_encoded(QByteArray(
-        reinterpret_cast<const char*>(fbb.GetBufferPointer()), fbb.GetSize()));
+    auto& rate_info = rate_limits[to_topic];
+    auto curr_time = std::chrono::high_resolution_clock::now();
```

Comment:
Just to be specific, `const auto`. If you mark as const and accidentally modify later, the compiler will find the bug for you.